app/models/template_input.rb

```diff
@@ -95,7 +103,13 @@ def ready?
     end
 
     def resolved_value
-      input_value.value
+      value = input_value.value
+
+      if @input.options_array.blank? || @input.options_array.include?(value)
+        value
+      else
+        raise(ValueNotPermitted, "Value '#{value}' is not a valid selection for #{@input.name}")
```

Comment:
I should probably just validate at creation of the TemplateInput, which I can do with a normal rails callback.

I put it here to validate at run time, but maybe that's not a good idea.  Do we do anything if the admin changes the provided values? Maybe the admin gives a user limited access to run job X with values A, B, and C, and later decides to not let them do C anymore.  Saved jobs would still let the user do C.  Maybe that's OK.

I guess we have that problem with everything, maybe they have saved jobs running on hosts they no longer have access to.